Source/Diagnostics/FullDiagnostics.cpp

```diff
@@ -634,7 +634,7 @@ FullDiagnostics::InitializeFieldFunctors (int lev)
         std::string varname = m_pfield_varnames[pcomp];
         for (int ispec=0; ispec<int(m_pfield_species.size()); ispec++) {
             m_all_field_functors[lev][nvar + pcomp * nspec + ispec] = std::make_unique<ParticleReductionFunctor>(nullptr,
-                    lev, m_crse_ratio, m_pfield_strings[varname], m_pfield_species_index[ispec],
+                    lev, m_crse_ratio, m_pfield_strings[varname], m_pfield_species_index[ispec], m_pfield_do_average[pcomp],
```

Comment:
Since `m_pfield_dofilter` and `m_pfield_filter_strings` are both `std::map`, would it make sense for `m_pfield_do_average` to also be, so the access pattern is consistent here?

Reply:
Yes, I can make it a map. Is there a reason that maps are used here for the other quantities? It doesn't seem like that is needed since a vector would work just as well. Are there plans for further changes the need maps?

Reply:
Yeah, I agree that maps and vectors are essentially interchangeable here. I think I originally had it as a map to avoid bookkeeping errors, since I initially had a separate vector of functors for these diagnostics, instead of appending them to `m_all_field_functors`. I don't have any plans that would require keeping them as maps, though.

Reply:
Ok. For the sake of code simplification, would it be ok if I change the maps to be vectors?

Reply:
Sure, go for it.

Reply:
Changes done.

BTW, I think that this diagnostic will work with RZ, depositing the data on a r-z grid. Since the code is accessing the `SuperParticleType` directly, the `p.pos(0)` will be `r` and `p.pos(1)` will be `z`. The filtering parser however will work with `x`, `y`, and `z`.

Reply:
Awesome. Thanks! You're probably right, and if it doesn't work it would likely be a quick fix. Though some additional code would need to be added in either `Diagnostics.cpp` or `FullDiagnostics.cpp` to create the functors when RZ is turned on. It may work with plotfiles but it definitely doesn't work with RZ+OpenPMD at the moment. I'm also not sure if we'd also want to output modes for these fields, or if that's just for the fields used in evolution.

Reply:
Yes, some extra work will be needed, and there are complications. I don't need this now in RZ. This can be done in a future PR.

Reply:
Thanks for the comments!